daemon/build.go

```diff
@@ -75,20 +76,30 @@ func (rl *releaseableLayer) Release() error {
 	if rl.released {
 		return nil
 	}
+	if err := rl.releaseRWLayer(); err != nil {
+		return err
+	}
+	if err := rl.releaseROLayer(); err != nil {
+		return err
+	}
 	rl.released = true
-	rl.releaseRWLayer()
-	return rl.releaseROLayer()
+	return nil
 }
 
 func (rl *releaseableLayer) releaseRWLayer() error {
 	if rl.rwLayer == nil {
 		return nil
 	}
+	if err := rl.rwLayer.Unmount(); err != nil {
```

Comment:
In line 41, when `Mount()` returns error should release `rwlayer` (maybe would get released anyway but not good idea to leak when function fails). At least it makes sure that we never unmount when we have not mounted before.